python/pyiceberg/utils/schema_conversion.py

```diff
@@ -592,13 +599,13 @@ def visit_timestamp(self, timestamp_type: TimestampType) -> AvroType:
 
     def visit_timestamptz(self, timestamptz_type: TimestamptzType) -> AvroType:
         # Iceberg only supports micro's
-        return {"type": "long", "logicalType": "timestamp-micros"}
+        return {"type": "long", "logicalType": "timestamp-micros", "adjust-to-utc": True}
```

Comment:
The spec shows that this is also explicitly set for the `timestamp` type: `"adjust-to-utc": false`